multi-node/aws/pkg/cluster/cluster.go

```diff
@@ -42,38 +45,132 @@ func New(cfg *config.Cluster, awsDebug bool) *Cluster {
 	}
 
 	return &Cluster{
-		clusterName: aws.String(cfg.ClusterName),
-		svc:         cloudformation.New(session.New(awsConfig)),
+		Cluster: *cfg,
+		session: session.New(awsConfig),
 	}
 }
 
 type Cluster struct {
-	clusterName *string
-	svc         *cloudformation.CloudFormation
+	config.Cluster
+	session *session.Session
 }
 
 func (c *Cluster) ValidateStack(stackBody string) (string, error) {
-	input := &cloudformation.ValidateTemplateInput{
+	validateInput := cloudformation.ValidateTemplateInput{
 		TemplateBody: &stackBody,
 	}
 
-	validationReport, err := c.svc.ValidateTemplate(input)
+	cfSvc := cloudformation.New(c.session)
+	validationReport, err := cfSvc.ValidateTemplate(&validateInput)
 	if err != nil {
-		return "", fmt.Errorf("Invalid cloudformation stack: %v", err)
+		return "", fmt.Errorf("invalid cloudformation stack: %v", err)
 	}
 
-	return validationReport.String(), err
+	describeStacksInput := cloudformation.DescribeStacksInput{
+		StackName: aws.String(c.ClusterName),
+	}
+
+	//Find out if stack exists already. This determines whether we should do subnet conflict validatio
+	var stackExists bool
+	stackNotExistExpr := regexp.MustCompile(fmt.Sprintf("^ValidationError: Stack with id %s does not exist", c.ClusterName))
```

Comment:
`DescribeStacks` returns an Validation error if there are no stacks that have the given name.

The other option is to use `ListStacks` and loop through all the stacks checking for that stack name.